File: javaLabs/src/OtherGoods.java

public class OtherGoods extends Goods {
	OtherGoodsType otherGoodsType;

	public OtherGoods(String name, double price, int amount, ColorType color, CustumerType grade,
			OtherGoodsType otherGoodsType) {
		super(name, price, amount, color, grade);
		setOtherGoodsType(otherGoodsType);
	}

	public OtherGoodsType getOtherGoodsType() {
		return otherGoodsType;
	}

	public void setOtherGoodsType(OtherGoodsType otherGoodsType) {
		this.otherGoodsType = otherGoodsType;
	}

	@Override
	public String toString() {
		return super.toString() + " otherGoodsType = " + otherGoodsType + "\n";
	}

}

File: python/GShop/Goods/OtherGoods.py
from Goods.Goods import Goods
from enums.OtherGoodsType import OtherGoodsType


class OtherGoods(Goods):
    other_goods_type = OtherGoodsType

    def __init__(self, name, price, amount, custumer_type, color, other_goods_type):
        self.other_goods_type = other_goods_type

        Goods.__init__(self, name, price, amount, custumer_type, color)

    def __str__(self):
        return "Name : " + str(self.name) + " price: " + str(self.price) + "amount: " + str(self.amount) \
               + "color: " + str(self.color) + "custumer type: " + str(self.custumer_type) \
               + "goods type: " + str(self.other_goods_type)